File: go/isAnagram.go
// https://leetcode.com/problems/valid-anagram

package leetcode

func isAnagram(s string, t string) bool {
	seen := make([]int, 26)

	if len(s) != len(t) {
		return false
	}

	for _, val := range s {
		seen[val-'a']++
	}

	for _, val := range t {
		seen[val-'a']--
	}

	for _, val := range seen {
		if val != 0 {
			return false
		}
	}

	return true
}

File: c/src/isAnagram.c
// https://leetcode.com/problems/valid-anagram

#include <stdbool.h>
#include <strings.h>

bool isAnagram(char* s, char* t)
{
    int seen[26] = { 0 };

    for (; *s != '\0'; s++)
        seen[*s - 'a']++;

    for (; *t != '\0'; t++)
        seen[*t - 'a']--;

    for (size_t i = 0; i < 26; i++) {
        if (seen[i] != 0)
            return false;
    }

    return true;
}
